File: src/algorithms/add-digits.js
/**
 * Repeatedly adds all the digits of num until the result has only one digit
 * and returns that number.
 * 
 * @param {number} num - A number parameter
 * @returns {number} - Returns a single digit number
 */
function addDigits (num) {
  let sum = 0;
  let temp = num;
  while (String(num).length > 1) {
    while (temp !== 0) {
      sum += temp % 10;
      temp = Math.floor(temp / 10);
    }
    num = sum;
    temp = num;
    sum = 0;
  }
  return num;
}

File: src/algorithms/add-digits.py
def addDigits(num):
    """Given an integer num, repeatedly add all its digits until the result has 
    only one digit, and return it.

    Args:
        num (int): An integer greater than or equal to 0.

    Returns:
        int: Returns the sum of num's digits until the result has only one 
        digit.
    """
    if num == 0:
        return 0
    if num % 9 == 0:
        return 9
    else:
        return num % 9
